examples/qaoa.py

```diff
@@ -0,0 +1,166 @@
+  │           │           │           Rx(-0.448π) │           Rx(-0.448π)
+  │           │           │           │           │           │
+  M('m')──────M───────────M───────────M───────────M───────────M
+  │           │           │           │           │           │
+Optimizing objective function ...
+The largest cut value found was 7.
+The largest possible cut has size 7.
+The approximation ratio achieved is 1.0.
+"""
+
+import itertools
+
+import networkx
+import numpy as np
+import scipy.optimize
+
+import cirq
+
+
+def main():
+    # Set problem parameters
+    n = 6
+    p = 2
+
+    # Generate a random 3-regular graph on n nodes
+    graph = networkx.random_regular_graph(3, n)
+
+    # Make qubits
+    qubits = cirq.LineQubit.range(n)
+
+    # Print an example circuit
+    betas = np.random.uniform(-np.pi, np.pi, size=p)
+    gammas = np.random.uniform(-np.pi, np.pi, size=p)
+    circuit = qaoa_max_cut_circuit(qubits, betas, gammas, graph)
+    print('Example QAOA circuit:')
+    print(circuit.to_text_diagram(transpose=True))
+
+    # Create variables to store the largest cut and cut value found
+    largest_cut_found = None
+    largest_cut_value_found = 0
+
+    # Initialize simulator
+    simulator = cirq.Simulator()
+
+    # Define objective function (we'll use the negative expected cut value)
+    num_samples = 1000
+    def f(x):
+        # Create circuit
+        betas = x[:p]
+        gammas = x[p:]
+        circuit = qaoa_max_cut_circuit(qubits, betas, gammas, graph)
+        # Sample bitstrings from circuit
+        result = simulator.run(circuit, repetitions=num_samples)
+        bitstrings = result.measurements['m']
+        # Process bitstrings
+        sum_of_cut_values = 0
+        nonlocal largest_cut_found
+        nonlocal largest_cut_value_found
+        for bitstring in bitstrings:
+            value = cut_value(bitstring, graph)
+            sum_of_cut_values += value
+            if value > largest_cut_value_found:
+                largest_cut_value_found = value
+                largest_cut_found = bitstring
+        mean = sum_of_cut_values / num_samples
+        return -mean
+
+    # Pick an initial guess
+    x0 = np.random.uniform(-np.pi, np.pi, size=2*p)
+
+    # Optimize f
+    print('Optimizing objective function ...')
+    scipy.optimize.minimize(f,
+                            x0,
+                            method='Nelder-Mead',
+                            options={'maxiter': 50})
+
+    # Compute best possible cut value via brute force search
+    max_cut_value = max(cut_value(bitstring, graph)
+                        for bitstring in itertools.product(range(2), repeat=n))
+
+    # Print the results
+    print('The largest cut value found was {}.'.format(largest_cut_value_found))
+    print('The largest possible cut has size {}.'.format(max_cut_value))
+    print('The approximation ratio achieved is {}.'.format(
+        largest_cut_value_found / max_cut_value))
+
+
+def Rzz(rads):
```

Comment:
nit: style would say make this rzz

Reply:
We have cirq.Rz though

Reply:
cirq.Rz is a class.  Here you have a method.